src/client/components/forms/registration-details.js

```diff
@@ -104,6 +104,10 @@ class RegistrationForm extends React.Component {
 			width: '100%'
 		};
 
+		function birthdayValidation(value) {
```

Comment:
Instead of repeating here, this would be best split into a separate helper file.